mteb/evaluation/evaluators/SummarizationEvaluator.py

```diff
@@ -42,10 +42,18 @@ def __init__(
         self.machine_summaries = machine_summaries
         self.texts = texts
         self.gold_scores = gold_scores
-        self.batch_size = batch_size
         self.task_name = task_name
 
-    def __call__(self, model: Encoder | EncoderWithSimilarity):
+    def __call__(
+        self,
+        model: Encoder | EncoderWithSimilarity,
+        *,
+        encode_kwargs: dict[str, Any] = {},
+    ):
+        # set default for encode_kwargs
+        if "batch_size" not in encode_kwargs:
+            encode_kwargs["batch_size"] = 32
```

Comment:
I noticed that sometimes this if statement is in `__init__()` for some evaluators, and sometimes in `__call__()`. Since the tests are passing, we know the arg will be passed into `encode`, so we're good there. Just a note for us that this might be something we'd want to revisit later to standardize.

Reply:
Yes you are completely correct. Already partly standardized it, but already made quite large changes and wanted to keep it as simple as possible (especially for retrieval and reranking the classes have also grown excessively large, which makes it hard to debug)

Reply:
Makes sense!

Reply:
Perfect will that I will add it in then.